haskell/ghc_bindist.bzl

```diff
@@ -604,6 +604,14 @@ def haskell_register_ghc_bindists(
 
     [toolchain-resolution]: https://docs.bazel.build/versions/master/toolchains.html#toolchain-resolution
 
+    Args:
```

Comment:
`rules_haskell_toolchains` defined in `haskell/toolchains.bzl` is the entry point that is documented in all the examples and use-case docs. It itself then calls `haskell_register_ghc_bindists`. I think in this case it's better to use this opportunity and move this docstring and the argument docs from `ghc_bindist` to `rules_haskell_toolchains` and have those reference `rules_haskell_toolchains`.

Reply:
I figured that if users are loading the rule like this
`load("@rules_haskell//haskell:toolchain.bzl", "rules_haskell_toolchains")`
then I should expose the `toolchain.bzl` file in the documentation.

So now `haskell_toolchain` appears twice in the menu (since it is also reexported in `def.bzl`).

They are now three names for `rules_haskell_toolchains`.
 - `rules_haskell_toolchains` itself
 - `haskell_register_toolchains`
 - `haskell_register_ghc_bindists`
   
Are the last two depreciated ?

Reply:
Good catch. `haskell_register_toolchains` is deprecated, see [here in the changelog](https://github.com/tweag/rules_haskell/blob/377db34fec4342ee0c317725669334479b6ecc10/CHANGELOG.md#changed-3). So, that should be marked accordingly.

Re `haskell_register_ghc_bindists`, that's a good question. There's ongoing work on more toolchains, in particular a [host toolchain](https://github.com/tweag/rules_haskell/pull/1466/files). I'm not sure yet how this will affect `rules_haskell_toolchains`. I'd say for the purpose of this PR the link from `haskell_register_ghc_bindists` to `rules_haskell_toolchains` is good the way it is.

Reply:
Ok. I marked `haskell_register_toolchains` as deprecated.